examples/noipClient/noipClient.ino

```diff
@@ -48,6 +48,12 @@ static uint32_t request_timer;
 int attempt;
 Stash stash;
 
+void checkPublicIP();
```

Comment:
I have never had to declare functions with Arduino. Maybe because I normally define them before they are called.
I wonder if that would be better?

Reply:
> the (Arduino) IDE does do a certain amount of pre-processing (in particular, **generating function prototypes for you**) before submitting the code to the main C++ compiler

That's why you don't need to declare functions before they're used in `.ino` files.
I see no harm in having the prototypes in the file.

http://forum.arduino.cc/index.php?&topic=97455.msg731293#msg731293

Reply:
I think there's no need to move functions around just to avoid the prototypes.

@ArnaudD-FR I don't think gcc supports adding prototypes like the Arduino IDE does. Whatever tool you're using, it must generate function prototypes like the Arduino IDE does so it can handle "standard" `.ino` (before passing them to gcc). Are you using custom Makefiles? What's your build environment? Anything besides gcc?

Reply:
I did not mean gcc is adding prototypes automatically at the beginning, but gcc has maybe some flags to allow undefined functions to be called without any definitions like in C where you can use a function wihtout any previous declaration (a warning is raised in this case)

Here is my simple Makefile at the root of my rain tanker project, I use the same to build any example:
```makefile
BOARD_TAG = nano328
ARDUINO_LIBS = EtherCard
USER_LIB_PATH = libraries

CFLAGS+=-Wall -Wextra
CXXFLAGS+=-Wall -Wextra

ifeq ($(DEV),Y)
    CFLAGS+=-DTANK_DEV
    CXXFLAGS+=-DTANK_DEV
    BOARD_SUB+=dev
endif

include /usr/share/arduino/Arduino.mk
```